feat(curriculum): Add set 2 of questions for CSS Flexbox quiz (#61067)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md

Fixed sizing and space between items like a grid.

## --quiz--

### --question--

> #### --text--
> What is the purpose of a one-dimensional layout model like Flexbox?

Reviewer: please double check your formatting, it needs to follow exactly the schema
```suggestion
#### --text--

What is the purpose of a one-dimensional layout model like Flexbox?
```
```suggestion
#### --text--
What is the purpose of a one-dimensional layout model like Flexbox?
```

you should not miss empty lines

#### --distractors--

```css
.container {
  display: flex;
  flex-flow: row wrap;
}
```

---

```css
.container {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
}
```

---

```css
.container {
  display: flex;
  justify-content: center;
  align-items: center;
}
```

#### --answer--

```css
.container {
  display: flex;
  flex-flow: column wrap;
}
```

### --question--

#### --text--

Which `justify-content` value aligns all flex items to the end of the container's main axis?

#### --distractors--

`justify-content: end;`

---

`justify-content: center;`

---

`justify-content: flex-start;`

#### --answer--

`justify-content: flex-end;`

### --question--

#### --text--

What is the effect of the following CSS rule on flex items?

```css
.container {
  flex-flow: row nowrap;
}
```

#### --distractors--

Items are arranged in a column and will wrap to a new column.

---

Items are arranged in a row and will wrap to a new row.

---

Items are arranged in a column and will overflow if they don't fit.

#### --answer--

Items are arranged in a row and will overflow the container if they don't fit.

### --question--

#### --text--

> Which CSS snippet will place the first flex item at the far left, the last item at the far right, and distribute the remaining items evenly in the space between them?

Reviewer 2: this question seemed a little long to me.

It would be better to ask what the result would be for the following code snippet and then the answers would be describing the outcome

#### --distractors--

```css
.container {
  display: flex;
  justify-content: space-around;
}
```

---

```css
.container {
  display: flex;
  justify-content: center;
}
```

---

```css
.container {
  display: flex;
  justify-content: space-evenly;
}
```

#### --answer--

```css
.container {
  display: flex;
  justify-content: space-between;
}
```

### --question--

#### --text--

> If you set `flex-direction: row;`, which property would you use to center the items vertically?

Reviewer 2: ```suggestion
Which property would you use to vertically center items when `flex-direction: row;` is set?
```

  align-items: flex-start;
}
```

---

```css
.container {
  display: flex;
  flex-wrap: wrap-reverse;
  align-items: flex-start;
}
```

#### --answer--

```css
.container {
  display: flex;
  flex-direction: row-reverse;
  align-items: flex-start;
}
```

### --question--

#### --text--

Which of the following is a valid declaration for the `flex-flow` property?

#### --distractors--

`flex-flow: center wrap;`

---

`flex-flow: row space-between;`

---

`flex-flow: column-reverse nowrap;`

#### --answer--

`flex-flow: column wrap-reverse;`

### --question--

#### --text--

Which of the following CSS rules will result in flex items being centered horizontally and stretched vertically to fill the height of the container?

#### --distractors--

```css
.container {
  display: flex;
  justify-content: center;
  align-items: center;
}
```

---

```css
.container {
  display: flex;
  justify-content: stretch;
  align-items: center;
}
```

---

```css
.container {
  display: flex;
  justify-content: flex-start;
  align-items: stretch;
}
```

#### --answer--

```css
.container {
  display: flex;
  justify-content: center;
  align-items: stretch;
}
```

### --question--

#### --text--

> If `flex-direction` is `column`, what does `align-items: stretch;` do?

Reviewer 2: ```suggestion
What effect does `align-items: stretch;` have when `flex-direction` is set to `column`?
```
